server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/SpanServiceElasticsearch.java

```diff
@@ -0,0 +1,212 @@
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * @author Pavol Loffay
+ */
+public class SpanServiceElasticsearch implements SpanService {
+
+    private static final MsgLogger log = MsgLogger.LOGGER;
+
+    public static final String SPAN_TYPE = "span";
+
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    private ElasticsearchClient client = ElasticsearchClient.getSingleton();
+
+
+    @Override
+    public Span getSpan(String tenantId, String id) {
+
+        GetResponse response = null;
+        try {
+            response = client.getClient()
+                    .prepareGet(client.getIndex(tenantId), SPAN_TYPE, id)
+                    .setRouting(id)
+                    .execute()
+                    .actionGet();
+        } catch (IndexMissingException ex) {
+            log.errorf("Missing span index %s", tenantId);
+            return null;
+        }
+
+        Span span = null;
+        if (!response.isSourceEmpty()) {
+            try {
+                span = deserialize(response.getSourceAsString(), Span.class);
+            } catch (IOException ex) {
+                log.errorFailedToParse(ex);
+                throw new IllegalStateException("Failed to deserialize span", ex);
+            }
+        }
+
+        log.tracef("Get span with id[%s] is: %s", id, span);
+
+        return span;
+    }
+
+    @Override
+    public List<Span> getChildren(String tenantId, String id) {
+
+        List<Span> spans = new ArrayList<>();
+        final String index = client.getIndex(tenantId);
+        try {
+
+            RefreshRequestBuilder refreshRequestBuilder = client.getClient()
+                    .admin()
+                    .indices()
+                    .prepareRefresh(index);
+
+            client.getClient()
+                    .admin()
+                    .indices()
+                    .refresh(refreshRequestBuilder.request())
+                    .actionGet();
+
+            QueryBuilder query = QueryBuilders.termQuery("parentId", id);
+
+            SearchRequestBuilder request = client.getClient()
+                    .prepareSearch(index)
+                    .setTypes(SPAN_TYPE)
+                    .setSearchType(SearchType.DFS_QUERY_THEN_FETCH)
+                    .setQuery(query);
+
+            SearchResponse response = request.execute()
+                    .actionGet();
+
+            for (SearchHit searchHitFields : response.getHits()) {
+                try {
+                    spans.add(deserialize(searchHitFields.getSourceAsString(), Span.class));
+                } catch (IOException ex) {
+                    log.errorFailedToParse(ex);
+                    throw new IllegalStateException("Failed to deserialize span", ex);
+                }
+            }
+        } catch (IndexMissingException ex) {
+            log.errorf("No index[%s] found, so unable to retrieve spans", index);
```

Comment:
The semantics of the method can be defined by you, but I'd rather return a `null`, with the semantic that `the index doesn't exist, so, I'm giving you a null return`, and an empty list as a signal for `I found the index, but there's no data stored for it yet`.